server/src/main/java/org/elasticsearch/rest/MethodHandlers.java

```diff
@@ -20,28 +21,23 @@
 final class MethodHandlers {
 
     private final String path;
-    private final Map<RestRequest.Method, Map<RestApiCompatibleVersion, RestHandler>> methodHandlers;
+    private final Map<Method, Map<RestApiCompatibleVersion, RestHandler>> methodHandlers;
 
-    MethodHandlers(String path, RestHandler handler, RestRequest.Method... methods) {
+    MethodHandlers(String path) {
         this.path = path;
-        this.methodHandlers = new HashMap<>(methods.length);
-        for (RestRequest.Method method : methods) {
-            methodHandlers.computeIfAbsent(method, k -> new HashMap<>())
-                .put(handler.compatibleWithVersion(), handler);
-        }
+        this.methodHandlers = new HashMap<>(1);
```

Comment:
isn't it common that we have multiple methods per same endpoint?

Reply:
Yes, but the default capacity for a HashMap is 16, and we create about 600 MethodHandlers in the course of starting Elasticsearch. ~450 of those have just one Route in them. ~100 have 2, and the rest have more (a few have as many as five).

It seemed to me like the cost of the HashMap resizing was worth the memory saving, but I ran a little bit of analysis and I'm going to add a commit that bumps the default initial capacity to 4.

Reply:
0b50128